src/service/apiService.js

```diff
@@ -0,0 +1,30 @@
+import axios from "axios";
+
+axios.defaults.baseURL = "https://spares-backend-i2mq.onrender.com/api/";
+
+const setAuthHeader = (token) => {
+  axios.defaults.headers.common.Authorization = `Bearer ${token}`;
+};
+
+const clearAuthHeader = () => {
+  axios.defaults.headers.common.Authorization = "";
+};
+
+//operations with users
+
+export const register = async (credentials) => {
+  const { data } = await axios.post("users/register", credentials);
+  setAuthHeader(data.token);
```

Comment:
all the points which depend on side sources (here it`s backed), should be covered byt try...catch... or ?.

here, if the response is unsuccessfull, you might have the mistake "can`t take property 'token' of underfined"

Reply:
not always it'll you mistake, but try...catch... helps to prevent outside mistakes and your front part won`t be crashed